transmart-rest-api-e2e/src/test/groovy/base/RestHelper.groovy

```diff
@@ -17,6 +17,8 @@ class RestHelper {
 
             if (!requestMap.skipOauth && AUTH_NEEDED) {
                 testContext.getAuthAdapter().authenticate(getRequest(), (requestMap.user ?: DEFAULT_USER))
+            } else if (requestMap.skipOauth && requestMap.token){
```

Comment:
Could we perhaps simplify the conditions by reducing the number of flags?
Are `skipOauth` and `AUTH_NEEDED` always opposite flags to each other? When they not?
Are `requestMap.skipOauth` and `requestMap.token` always have the same boolean value? When they not?

Reply:
`requestMap.token` is a string, but I can replace the skipOauth one with it, because `skipOauth` is not used anymore in other cases that I added.

Regarding `AUTH_NEEDED` it was used for the previous flow with transmart-oauth2 - there was an option to switch off the authentication in the configuration file. Since we are still supporting authentication with transmart for tm-server, we should probably keep this `AUTH_NEEDED`. What do you think?